Naveed Khan

# COVID-19 Clinical Trials: Exploratory Data Analysis (EDA)

# Question 1:
**Read Dataset and Explore the dataset by checking shape, columns, see the first/last 'n' rows using head/tail. (n= 5,15,30)**

In [ ]:
import pandas as pd

In [ ]:
df = pd.read_csv('C:\\Users\\92311\\Downloads\\COVID clinical trials.csv')

In [ ]:
df



Rank   NCT Number                                              Title  \
0        1  NCT04785898  Diagnostic Performance of the ID Now™ COVID-19...   
1        2  NCT04595136  Study to Evaluate the Efficacy of COVID19-0001...   
2        3  NCT04395482  Lung CT Scan Analysis of SARS-CoV2 Induced Lun...   
3        4  NCT04416061  The Role of a Private Hospital in Hong Kong Am...   
4        5  NCT04395924         Maternal-foetal Transmission of SARS-Cov-2   
...    ...          ...                                                ...   
5778  5779  NCT04011644  Mobile Health for Alcohol Use Disorders in Cli...   
5779  5780  NCT04681339  Antibiotic Prescription in Children Hospitaliz...   
5780  5781  NCT04740229  Moderate-intensity Flow-based Yoga Effects on ...   
5781  5782  NCT04804917           3-year Follow-up of the Mind My Mind RCT   
5782  5783  NCT04680000  Chronic Pain Management In Primary Care Using ...   

           Acronym                  Status         Study Results  \
0      COVID-IDNow  Active, not recruiting  No Results Available   
1         COVID-19      Not yet recruiting  No Results Available   
2      TAC-COVID19              Recruiting  No Results Available   
3         COVID-19  Active, not recruiting  No Results Available   
4     TMF-COVID-19              Recruiting  No Results Available   
...            ...                     ...                   ...   
5778           NaN              Recruiting  No Results Available   
5779           NaN      Not yet recruiting  No Results Available   
5780           NaN              Recruiting  No Results Available   
5781  MindMyMindFU              Recruiting  No Results Available   
5782           NaN      Not yet recruiting  No Results Available   

                                             Conditions  \
0                                               Covid19   
1                                  SARS-CoV-2 Infection   
2                                               covid19   
3                                                 COVID   
4     Maternal Fetal Infection Transmission|COVID-19...   
...                                                 ...   
5778                      Alcohol Drinking|Telemedicine   
5779  Community Acquired Pneumonia in Children|Antib...   
5780                               Stress|Psychological   
5781  Emotional Problem|Anxiety Disorder of Childhoo...   
5782                                       Chronic Pain   

                                          Interventions  \
0      Diagnostic Test: ID Now™ COVID-19 Screening Test   
1       Drug: Drug COVID19-0001-USR|Drug: normal saline   
2     Other: Lung CT scan analysis in COVID-19 patients   
3             Diagnostic Test: COVID 19 Diagnostic Test   
4     Diagnostic Test: Diagnosis of SARS-Cov2 by RT-...   
...                                                 ...   
5778  Behavioral: A-CHESS self-monitored|Behavioral:...   
5779  Other: Antibiotic treatment|Other: No antibiot...   
5780                                   Behavioral: Yoga   
5781                                                NaN   
5782  Behavioral: Brief Cognitive Behavioral Therapy...   

                                       Outcome Measures  \
0     Evaluate the diagnostic performance of the ID ...   
1     Change on viral load results from baseline aft...   
2     A qualitative analysis of parenchymal lung dam...   
3     Proportion of asymptomatic subjects|Proportion...   
4     COVID-19 by positive PCR in cord blood and / o...   
...                                                 ...   
5778  Number of risky drinking days|Number of patien...   
5779  Antibiotic treatment rates in hospitalized chi...   
5780  Perceived Stress|Task switching paradigm|Digit...   
5781  The child's impact of mental health problems r...   
5782  Defense and Veterans Pain Rating Scale (DVPRS)...   

                                  Sponsor/Collaborators  ...  \
0                 Groupe Hospitalier Paris Saint Joseph  ... 

In [ ]:
df.T

0     \
Rank                                                                     1   
NCT Number                                                     NCT04785898   
Title                    Diagnostic Performance of the ID Now™ COVID-19...   
Acronym                                                        COVID-IDNow   
Status                                              Active, not recruiting   
Study Results                                         No Results Available   
Conditions                                                         Covid19   
Interventions             Diagnostic Test: ID Now™ COVID-19 Screening Test   
Outcome Measures         Evaluate the diagnostic performance of the ID ...   
Sponsor/Collaborators                Groupe Hospitalier Paris Saint Joseph   
Gender                                                                 All   
Age                              18 Years and older   (Adult, Older Adult)   
Phases                                                      Not Applicable   
Enrollment                                                          1000.0   
Funded Bys                                                           Other   
Study Type                                                  Interventional   
Study Designs            Allocation: N/A|Intervention Model: Single Gro...   
Other IDs                                                      COVID-IDNow   
Start Date                                                November 9, 2020   
Primary Completion Date                                  December 22, 2020   
Completion Date                                             April 30, 2021   
First Posted                                                 March 8, 2021   
Results First Posted                                                   NaN   
Last Update Posted                                           March 8, 2021   
Locations                Groupe Hospitalier Paris Saint-Joseph, Paris, ...   
Study Documents                                                        NaN   
URL                            https://ClinicalTrials.gov/show/NCT04785898   

                                                                      1     \
Rank                                                                     2   
NCT Number                                                     NCT04595136   
Title                    Study to Evaluate the Efficacy of COVID19-0001...   
Acronym                                                           COVID-19   
Status                                                  Not yet recruiting   
Study Results                                         No Results Available   
Conditions                                            SARS-CoV-2 Infection   
Interventions              Drug: Drug COVID19-0001-USR|Drug: normal saline   
Outcome Measures         Change on viral load results from baseline aft...   
Sponsor/Collaborators                           United Medical Specialties   
Gender                                                                 All   
Age                              18 Years and older   (Adult, Older Adult)   
Phases                                                     Phase 1|Phase 2   
Enrollment                                                            60.0   
Funded Bys                                                           Other   
Study Type                                                  Interventional   
Study Designs            Allocation: Randomized|Intervention Model: Par...   
Other IDs                                                 COVID19-0001-USR   
Start Date                                                November 2, 2020   
Primary Completion Date                                  December 15, 2020   
Completion Date                                           January 29, 2021   
First Posted                                              October 20, 2020   
Results First Posted                                                   NaN   
Last Update P

In [ ]:
df.shape

(5783, 27)

In [ ]:
df.columns

Index(['Rank', 'NCT Number', 'Title', 'Acronym', 'Status', 'Study Results',
       'Conditions', 'Interventions', 'Outcome Measures',
       'Sponsor/Collaborators', 'Gender', 'Age', 'Phases', 'Enrollment',
       'Funded Bys', 'Study Type', 'Study Designs', 'Other IDs', 'Start Date',
       'Primary Completion Date', 'Completion Date', 'First Posted',
       'Results First Posted', 'Last Update Posted', 'Locations',
       'Study Documents', 'URL'],
      dtype='object')

**See the first/last 'n' rows using head/tail. (n= 5,15,30)**

In [ ]:
df.head(5)

Rank   NCT Number                                              Title  \
0     1  NCT04785898  Diagnostic Performance of the ID Now™ COVID-19...   
1     2  NCT04595136  Study to Evaluate the Efficacy of COVID19-0001...   
2     3  NCT04395482  Lung CT Scan Analysis of SARS-CoV2 Induced Lun...   
3     4  NCT04416061  The Role of a Private Hospital in Hong Kong Am...   
4     5  NCT04395924         Maternal-foetal Transmission of SARS-Cov-2   

        Acronym                  Status         Study Results  \
0   COVID-IDNow  Active, not recruiting  No Results Available   
1      COVID-19      Not yet recruiting  No Results Available   
2   TAC-COVID19              Recruiting  No Results Available   
3      COVID-19  Active, not recruiting  No Results Available   
4  TMF-COVID-19              Recruiting  No Results Available   

                                          Conditions  \
0                                            Covid19   
1                               SARS-CoV-2 Infection   
2                                            covid19   
3                                              COVID   
4  Maternal Fetal Infection Transmission|COVID-19...   

                                       Interventions  \
0   Diagnostic Test: ID Now™ COVID-19 Screening Test   
1    Drug: Drug COVID19-0001-USR|Drug: normal saline   
2  Other: Lung CT scan analysis in COVID-19 patients   
3          Diagnostic Test: COVID 19 Diagnostic Test   
4  Diagnostic Test: Diagnosis of SARS-Cov2 by RT-...   

                                    Outcome Measures  \
0  Evaluate the diagnostic performance of the ID ...   
1  Change on viral load results from baseline aft...   
2  A qualitative analysis of parenchymal lung dam...   
3  Proportion of asymptomatic subjects|Proportion...   
4  COVID-19 by positive PCR in cord blood and / o...   

                               Sponsor/Collaborators  ...         Other IDs  \
0              Groupe Hospitalier Paris Saint Joseph  ...       COVID-IDNow   
1                         United Medical Specialties  ...  COVID19-0001-USR   
2                       University of Milano Bicocca  ...       TAC-COVID19   
3                    Hong Kong Sanatorium & Hospital  ...        RC-2020-08   
4  Centre Hospitalier Régional d'Orléans|Centre d...  ...      CHRO-2020-10   

         Start Date Primary Completion Date   Completion Date  \
0  November 9, 2020       December 22, 2020    April 30, 2021   
1  November 2, 2020       December 15, 2020  January 29, 2021   
2       May 7, 2020           June 15, 2021     June 15, 2021   
3      May 25, 2020           July 31, 2020   August 31, 2020   
4       May 5, 2020                May 2021          May 2021   

       First Posted Results First Posted Last Update Posted  \
0     March 8, 2021                  NaN      March 8, 2021   
1  October 20, 2020                  NaN   October 20, 2020   
2      May 20, 2020                  NaN   November 9, 2020   
3      June 4, 2020                  NaN       June 4, 2020   
4      May 20, 2020                  NaN       June 4, 2020   

                                           Locations Study Documents  \
0  Groupe Hospitalier Paris Saint-Joseph, Paris, ...             NaN   
1       Cimedical, Barranquilla, Atlantico, Colombia             NaN   
2  Ospedale Papa Giovanni XXIII, Bergamo, Italy|P...             NaN   
3  Hong Kong Sanatorium & Hospital, Hong Kong, Ho...             NaN   
4                       CHR Orléans, Orléans, France             NaN   

                                           URL  
0  https://ClinicalTrials.gov/show/NCT04785898  
1  https://ClinicalTrials.gov/show/NCT04595136  
2  https://ClinicalTrials.gov/show/NCT04395482  
3  https://ClinicalTrials.gov/show/NCT04416061  
4  https://ClinicalTrials.gov/show/NCT04395924  

[5 rows x 27 columns]

In [ ]:
df.head(15)

Rank   NCT Number                                              Title  \
0      1  NCT04785898  Diagnostic Performance of the ID Now™ COVID-19...   
1      2  NCT04595136  Study to Evaluate the Efficacy of COVID19-0001...   
2      3  NCT04395482  Lung CT Scan Analysis of SARS-CoV2 Induced Lun...   
3      4  NCT04416061  The Role of a Private Hospital in Hong Kong Am...   
4      5  NCT04395924         Maternal-foetal Transmission of SARS-Cov-2   
5      6  NCT04516954          Convalescent Plasma for COVID-19 Patients   
6      7  NCT04476940  COVID-19 Breastfeeding Guideline for African-A...   
7      8  NCT04634214  The Severity of COVID 19 in Diabetes and Non-d...   
8      9  NCT04602884  Early Detection of COVID-19 Using Breath Analysis   
9     10  NCT04384588  COVID19-Convalescent Plasma for Treating Patie...   
10    11  NCT04355897  CoVID-19 Plasma in Treatment of COVID-19 Patients   
11    12  NCT04412265          Frailty in Elderly Patients With COVID-19   
12    13  NCT04659759  COVID-19 Pregnancy Related Immunological, Clin...   
13    14  NCT04427332     Smell and Taste Disorders in COVID-19 Patients   
14    15  NCT04842708  Evaluation of Anti-COVID 19 Pfizer Vaccination...   

         Acronym                   Status         Study Results  \
0    COVID-IDNow   Active, not recruiting  No Results Available   
1       COVID-19       Not yet recruiting  No Results Available   
2    TAC-COVID19               Recruiting  No Results Available   
3       COVID-19   Active, not recruiting  No Results Available   
4   TMF-COVID-19               Recruiting  No Results Available   
5           CPCP  Enrolling by invitation  No Results Available   
6       COVID-BF       Not yet recruiting  No Results Available   
7        COVID19       Not yet recruiting  No Results Available   
8       COVID-19                Suspended  No Results Available   
9     FALP-COVID               Recruiting  No Results Available   
10           NaN               Recruiting  No Results Available   
11     FRA-COVID               Recruiting  No Results Available   
12   COVID-PRICE               Recruiting  No Results Available   
13  COVID-19 ORL                Completed  No Results Available   
14      COVID-19               Recruiting  No Results Available   

                                           Conditions  \
0                                             Covid19   
1                                SARS-CoV-2 Infection   
2                                             covid19   
3                                               COVID   
4   Maternal Fetal Infection Transmission|COVID-19...   
5                                            COVID 19   
6                     Covid19|Exclusive Breastfeeding   
7                              Covid19|Type2 Diabetes   
8                                             Covid19   
9   COVID-19 Infection|Cancer Patients|General Pop...   
10                                           COVID 19   
11                                            Covid19   
12                                            Covid19   
13                                            covid19   
14                                            Covid19   

                                        Interventions  \
0    Diagnostic Test: ID Now™ COVID-19 Screening Test   
1     Drug: Drug COVID19-0001-USR|Drug: normal saline   
2   Other: Lung CT scan analysis in COVID-19 patients   
3           Diagnostic Test: COVID 19 Diagnostic Test   
4   Diagnostic Test: Diagnosis of SARS-Cov2 by RT-...   
5            Biological: Convalescent COVID 19 Plasma   
6          Behavioral: COVID-19 Breastfeeding Support   
7                                                 NaN   
8   Diagnostic Test: Breath biopsy sampling using ...   
9   Biological: Convalescent Plasma from COVID-19 ...   
10           Biological: Convalescent COVID 19 Plasma   
11  Other: Relation between frailty and clinical o...   
12  Other: COVID-19 exposure|Biological: COVID-19 ...   
13 

In [ ]:
df.head(30)

Rank   NCT Number                                              Title  \
0      1  NCT04785898  Diagnostic Performance of the ID Now™ COVID-19...   
1      2  NCT04595136  Study to Evaluate the Efficacy of COVID19-0001...   
2      3  NCT04395482  Lung CT Scan Analysis of SARS-CoV2 Induced Lun...   
3      4  NCT04416061  The Role of a Private Hospital in Hong Kong Am...   
4      5  NCT04395924         Maternal-foetal Transmission of SARS-Cov-2   
5      6  NCT04516954          Convalescent Plasma for COVID-19 Patients   
6      7  NCT04476940  COVID-19 Breastfeeding Guideline for African-A...   
7      8  NCT04634214  The Severity of COVID 19 in Diabetes and Non-d...   
8      9  NCT04602884  Early Detection of COVID-19 Using Breath Analysis   
9     10  NCT04384588  COVID19-Convalescent Plasma for Treating Patie...   
10    11  NCT04355897  CoVID-19 Plasma in Treatment of COVID-19 Patients   
11    12  NCT04412265          Frailty in Elderly Patients With COVID-19   
12    13  NCT04659759  COVID-19 Pregnancy Related Immunological, Clin...   
13    14  NCT04427332     Smell and Taste Disorders in COVID-19 Patients   
14    15  NCT04842708  Evaluation of Anti-COVID 19 Pfizer Vaccination...   
15    16  NCT04367805  COVID-19 Infection in Patients With Hepatocell...   
16    17  NCT04609969  Diagnostic Performance Evaluation of a Novel S...   
17    18  NCT04817553  Impact of COVID-19 on the Clinical Outcomes an...   
18    19  NCT04632355  Musculoskeletal Pain in Patients With Covid-19...   
19    20  NCT04407585  Testing the Accuracy of a Digital Test to Diag...   
20    21  NCT04657510                    Femoral frACturEs and COVID-19.   
21    22  NCT04473170  Study Evaluating the Safety and Efficacy of Au...   
22    23  NCT04497298  Clinical Trial to Evaluate the Safety and Immu...   
23    24  NCT04549831     Genetic Bases of COVID-19 Clinical Variability   
24    25  NCT04706390  Comparison of the Immune Response to Natural C...   
25    26  NCT04632719  The MentalPlus® for Assessment and Rehabilitat...   
26    27  NCT04427345               Predictive Factors COVID-19 Patients   
27    28  NCT04424004  MURDOCK Cabarrus County COVID-19 Prevalence an...   
28    29  NCT04407494  Diagnostic Value of Patient - Reported and Cli...   
29    30  NCT04360811  Assessment of Obstetric, Fetal and Neonatal Ri...   

          Acronym                   Status         Study Results  \
0     COVID-IDNow   Active, not recruiting  No Results Available   
1        COVID-19       Not yet recruiting  No Results Available   
2     TAC-COVID19               Recruiting  No Results Available   
3        COVID-19   Active, not recruiting  No Results Available   
4    TMF-COVID-19               Recruiting  No Results Available   
5            CPCP  Enrolling by invitation  No Results Available   
6        COVID-BF       Not yet recruiting  No Results Available   
7         COVID19       Not yet recruiting  No Results Available   
8        COVID-19                Suspended  No Results Available   
9      FALP-COVID               Recruiting  No Results Available   
10            NaN               Recruiting  No Results Available   
11      FRA-COVID               Recruiting  No Results Available   
12    COVID-PRICE               Recruiting  No Results Available   
13   COVID-19 ORL                Completed  No Results Available   
14       COVID-19               Recruiting  No Results Available   
15  COVID19-CHIEF               Recruiting  No Results Available   
16     COVID-VIRO                Completed  No Results Available   
17     IgG4-COVID               Recruiting  No Results Available   
18   Covid19-Pain       Not yet recruiting  No Results Available   
19            NaN               Recruiting  No Results Available   
20  FACE COVID-19               Recruiting  No Results Available   
21   SENTAD-COVID                Completed  No Results Available   
22   COVID-19-101   Active, not recruiting  No Results Available   
23      GEN-COV

In [ ]:
df.tail(5)

Rank   NCT Number                                              Title  \
5778  5779  NCT04011644  Mobile Health for Alcohol Use Disorders in Cli...   
5779  5780  NCT04681339  Antibiotic Prescription in Children Hospitaliz...   
5780  5781  NCT04740229  Moderate-intensity Flow-based Yoga Effects on ...   
5781  5782  NCT04804917           3-year Follow-up of the Mind My Mind RCT   
5782  5783  NCT04680000  Chronic Pain Management In Primary Care Using ...   

           Acronym              Status         Study Results  \
5778           NaN          Recruiting  No Results Available   
5779           NaN  Not yet recruiting  No Results Available   
5780           NaN          Recruiting  No Results Available   
5781  MindMyMindFU          Recruiting  No Results Available   
5782           NaN  Not yet recruiting  No Results Available   

                                             Conditions  \
5778                      Alcohol Drinking|Telemedicine   
5779  Community Acquired Pneumonia in Children|Antib...   
5780                               Stress|Psychological   
5781  Emotional Problem|Anxiety Disorder of Childhoo...   
5782                                       Chronic Pain   

                                          Interventions  \
5778  Behavioral: A-CHESS self-monitored|Behavioral:...   
5779  Other: Antibiotic treatment|Other: No antibiot...   
5780                                   Behavioral: Yoga   
5781                                                NaN   
5782  Behavioral: Brief Cognitive Behavioral Therapy...   

                                       Outcome Measures  \
5778  Number of risky drinking days|Number of patien...   
5779  Antibiotic treatment rates in hospitalized chi...   
5780  Perceived Stress|Task switching paradigm|Digit...   
5781  The child's impact of mental health problems r...   
5782  Defense and Veterans Pain Rating Scale (DVPRS)...   

                                  Sponsor/Collaborators  ...  \
5778  University of Wisconsin, Madison|National Inst...  ...   
5779  ARCIM Institute Academic Research in Complemen...  ...   
5780         University of Illinois at Urbana-Champaign  ...   
5781  Mental Health Services in the Capital Region, ...  ...   
5782  The University of Texas Health Science Center ...  ...   

                                              Other IDs         Start Date  \
5778  2019-0337|R01AA024150|A532007|SMPH/FAMILY MED/...     March 23, 2020   
5779                                             PKA-03         April 2021   
5780                                              21584  February 10, 2021   
5781                         MHSCRDenmark, F-61502-03-1     March 22, 2021   
5782                                       HSC20200520H      February 2021   

     Primary Completion Date    Completion Date       First Posted  \
5778             August 2022         April 2023       July 8, 2019   
5779           November 2024      December 2024  December 23, 2020   
5780               July 2021          July 2021   February 5, 2021   
5781            May 31, 2022  December 31, 2022     March 18, 2021   
5782           February 2024      February 2025  December 22, 2020   

     Results First Posted Last Update Posted  \
5778                  NaN      April 2, 2021   
5779                  NaN  February 10, 2021   
5780                  NaN  February 24, 2021   
5781                  NaN      April 1, 2021   
5782                  NaN  December 22, 2020   

                                              Locations Study Documents  \
5778  UW Health at the American Center, Madison, Wis...             NaN   
5779  Die Filderklinik, Filderstadt, Baden-Württembe...             NaN   
5780  University of Illinois at Urbana-Champaign, Ur...             NaN   
5781  Mental Health Services in the Capital Region, ...             NaN   
5782  Uniformed Services University for the Health S...             NaN   

                                              URL  
5778  https://Clinic

In [ ]:
df.tail(15)

Rank   NCT Number                                              Title  \
5768  5769  NCT04734795  The Prevalence of Dysfunctional Breathing in C...   
5769  5770  NCT04190368  Team Clinic: Virtual Expansion of an Innovativ...   
5770  5771  NCT03392883  Scaling Up Science-based Mental Health Interve...   
5771  5772  NCT04301518  Prematurity Risk Assessment Combined With Clin...   
5772  5773  NCT04607902  Harnessing Network Science to Personalize Scal...   
5773  5774  NCT04639661  Predictors of Periodontal Outcomes Post-sanati...   
5774  5775  NCT04180709  CBT to Reduce Insomnia and Improve Social Reco...   
5775  5776  NCT04335643  Telehealth CBT for Adolescents and Young Adult...   
5776  5777  NCT04589377  Mindfulness to Mitigate Psychological Threat a...   
5777  5778  NCT04574466  Scaling-up Psychological Interventions With Sy...   
5778  5779  NCT04011644  Mobile Health for Alcohol Use Disorders in Cli...   
5779  5780  NCT04681339  Antibiotic Prescription in Children Hospitaliz...   
5780  5781  NCT04740229  Moderate-intensity Flow-based Yoga Effects on ...   
5781  5782  NCT04804917           3-year Follow-up of the Mind My Mind RCT   
5782  5783  NCT04680000  Chronic Pain Management In Primary Care Using ...   

           Acronym                   Status         Study Results  \
5768           NaN               Recruiting  No Results Available   
5769           NaN       Not yet recruiting  No Results Available   
5770         DIADA   Active, not recruiting  No Results Available   
5771         PRIME               Recruiting  No Results Available   
5772           NaN               Recruiting  No Results Available   
5773           NaN  Enrolling by invitation  No Results Available   
5774         CRISP               Recruiting  No Results Available   
5775          cSLE               Recruiting  No Results Available   
5776           NaN               Recruiting  No Results Available   
5777           NaN               Recruiting  No Results Available   
5778           NaN               Recruiting  No Results Available   
5779           NaN       Not yet recruiting  No Results Available   
5780           NaN               Recruiting  No Results Available   
5781  MindMyMindFU               Recruiting  No Results Available   
5782           NaN       Not yet recruiting  No Results Available   

                                             Conditions  \
5768  Dysfunctional Breathing|Asthma in Children|Asthma   
5769                                    Type 1 Diabetes   
5770                 Depression|Problematic Alcohol Use   
5771                        Preterm Labor|Preterm Birth   
5772                                         Depression   
5773            Periodontal Diseases|Periodontal Pocket   
5774                Psychotic Disorders|Psychosis|Sleep   
5775                       Systemic Lupus Erythematosus   
5776                                             Stress   
5777  Distress|PTSD|Anxiety|Depression|Trauma|Functi...   
5778                      Alcohol Drinking|Telemedicine   
5779  Community Acquired Pneumonia in Children|Antib...   
5780                               Stress|Psychological   
5781  Emotional Problem|Anxiety Disorder of Childhoo...   
5782                                       Chronic Pain   

                                          Interventions  \
5768                                                NaN   
5769       Other: Team Clinic Care|Other: Standard Care   
5770                                  Behavioral: Laddr   
5771            Other: Multimodal intervention strategy   
5772  Behavioral: Supportive Therapy SSI|Behavioral:...   
5773                                                NaN   
5774                                    Device: Sleepio   
5775                                  Behavioral: TEACH   
5776                   Behavioral: Mindfulness Training   
5777                Behavioral: Problem Management Plus   
5778  Behavioral: A-CHESS self-monitored|Behavioral:...   

In [ ]:
df.tail(30)

Rank   NCT Number                                              Title  \
5753  5754  NCT04646291  Retrospective Study on the Use, Efficiency, an...   
5754  5755  NCT04735406                        The MS-LINK™ Outcomes Study   
5755  5756  NCT02495753  Vaginal Cleansing Before Cesarean Delivery to ...   
5756  5757  NCT03465280  Airway Intervention Registry (AIR): Recurrent ...   
5757  5758  NCT04576377  Dynamics of the Immune Responses to Repeat Inf...   
5758  5759  NCT04713735  Impact of Lactoferrin, a Dietary Supplement, v...   
5759  5760  NCT03873467  Group Lifestyle Balance™ for Individuals With ...   
5760  5761  NCT03594734  Group Lifestyle Balance™ for Individuals With ...   
5761  5762  NCT03728257  Lung Transplant G0 (LTGO): Improving Self-Mana...   
5762  5763  NCT04782518       Personalizing Exercise for Parkinson Disease   
5763  5764  NCT04783454  Preventing Neck & Back Pain in Teleworking Off...   
5764  5765  NCT03823469  Evaluating the Impact of a Culinary Coaching T...   
5765  5766  NCT04429061  Reaching 90 90 90 in Adolescents in Zambia: Us...   
5766  5767  NCT04184895  A Study to Assess the Safety, Tolerability and...   
5767  5768  NCT03922334                        Navigating New Motherhood 2   
5768  5769  NCT04734795  The Prevalence of Dysfunctional Breathing in C...   
5769  5770  NCT04190368  Team Clinic: Virtual Expansion of an Innovativ...   
5770  5771  NCT03392883  Scaling Up Science-based Mental Health Interve...   
5771  5772  NCT04301518  Prematurity Risk Assessment Combined With Clin...   
5772  5773  NCT04607902  Harnessing Network Science to Personalize Scal...   
5773  5774  NCT04639661  Predictors of Periodontal Outcomes Post-sanati...   
5774  5775  NCT04180709  CBT to Reduce Insomnia and Improve Social Reco...   
5775  5776  NCT04335643  Telehealth CBT for Adolescents and Young Adult...   
5776  5777  NCT04589377  Mindfulness to Mitigate Psychological Threat a...   
5777  5778  NCT04574466  Scaling-up Psychological Interventions With Sy...   
5778  5779  NCT04011644  Mobile Health for Alcohol Use Disorders in Cli...   
5779  5780  NCT04681339  Antibiotic Prescription in Children Hospitaliz...   
5780  5781  NCT04740229  Moderate-intensity Flow-based Yoga Effects on ...   
5781  5782  NCT04804917           3-year Follow-up of the Mind My Mind RCT   
5782  5783  NCT04680000  Chronic Pain Management In Primary Care Using ...   

           Acronym                   Status         Study Results  \
5753           NaN                Completed  No Results Available   
5754           NaN       Not yet recruiting  No Results Available   
5755           NaN   Active, not recruiting  No Results Available   
5756       AIR:RRP               Recruiting  No Results Available   
5757         DRIVE               Recruiting  No Results Available   
5758           NaN               Recruiting  No Results Available   
5759       GLB-CVA   Active, not recruiting  No Results Available   
5760           NaN   Active, not recruiting  No Results Available   
5761          LTGO               Recruiting  No Results Available   
5762           PEP               Recruiting  No Results Available   
5763  Prevent@HOME               Recruiting  No Results Available   
5764           NaN               Recruiting  No Results Available   
5765        SKILLZ               Recruiting  No Results Available   
5766           NaN               Recruiting  No Results Available   
5767          NNM2               Recruiting  No Results Available   
5768           NaN               Recruiting  No Results Available   
5769           NaN       Not yet recruiting  No Results Available   
5770         DIADA   Active, not recruiting  No Results Available   
5771         PRIME               Recruiting  No Results Available   
5772           NaN               Recruiting  No Results Available   
5773           NaN  Enrolling by invitation  No Results Available   
5774         CRISP               Recruiting  No Results Available   


In [ ]:
print(df["Enrollment"].mean())



18319.48860671421


In [ ]:
print(df["Enrollment"].median())

170.0


In [ ]:
print(df["Enrollment"].mode())

0    100.0
Name: Enrollment, dtype: float64


# Question 2: -
**Extract and document key insights by exploring column data types and applying info anddescribe and write down your insights inside the markdown.**

In [ ]:
# View the data types of each column
print(df.dtypes)

Rank                         int64
NCT Number                  object
Title                       object
Acronym                     object
Status                      object
Study Results               object
Conditions                  object
Interventions               object
Outcome Measures            object
Sponsor/Collaborators       object
Gender                      object
Age                         object
Phases                      object
Enrollment                 float64
Funded Bys                  object
Study Type                  object
Study Designs               object
Other IDs                   object
Start Date                  object
Primary Completion Date     object
Completion Date             object
First Posted                object
Results First Posted        object
Last Update Posted          object
Locations                   object
Study Documents             object
URL                         object
dtype: object



--> The dataset primarily comprises textual data represented as object data types, encompassing information such as study titles, conditions, and interventions. Two numerical columns, 'Rank' and 'Enrollment', are exceptions, with 'Rank' being an integer type providing ordinal rankings of studies, while 'Enrollment' is a floating-point type indicating the number of participants enrolled in each study.

In [ ]:
df.describe()

Rank    Enrollment
count  5783.000000  5.749000e+03
mean   2892.000000  1.831949e+04
std    1669.552635  4.045437e+05
min       1.000000  0.000000e+00
25%    1446.500000  6.000000e+01
50%    2892.000000  1.700000e+02
75%    4337.500000  5.600000e+02
max    5783.000000  2.000000e+07

--> Insight:
The 'Rank' column spans from 1 to 5783, signifying the ordinal ranking of studies, with a median rank of 2892. Conversely, the 'Enrollment' column ranges from 0 to 20,000,000, showcasing a median enrollment of 170 participants and a diverse array of enrollment values, implying varying study magnitudes.

Within the dataset, the "Rank" column delineates the position of each study, with an average placement hovering around 2892 out of 5783. Concurrently, the "Enrollment" column delineates the number of participants in each study, fluctuating from 0 to a maximum of 20 million, with most studies accommodating approximately 170 participants. This discrepancy suggests that while some studies involve a sizable cohort, others have significantly fewer participants, or none at all.

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5783 entries, 0 to 5782
Data columns (total 27 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Rank                     5783 non-null   int64  
 1   NCT Number               5783 non-null   object 
 2   Title                    5783 non-null   object 
 3   Acronym                  2480 non-null   object 
 4   Status                   5783 non-null   object 
 5   Study Results            5783 non-null   object 
 6   Conditions               5783 non-null   object 
 7   Interventions            4897 non-null   object 
 8   Outcome Measures         5748 non-null   object 
 9   Sponsor/Collaborators    5783 non-null   object 
 10  Gender                   5773 non-null   object 
 11  Age                      5783 non-null   object 
 12  Phases                   3322 non-null   object 
 13  Enrollment               5749 non-null   float64
 14  Funded Bys              

-->Data Structure:
The dataset comprises 5783 rows and 27 columns. The majority of columns are of object data type (textual), with only two numeric columns ('Rank' and 'Enrollment').


-->Missing Values:
Columns like 'Acronym', 'Interventions', 'Phases', 'Outcome Measures', and 'Study Documents' exhibit notable missing values, suggesting potential areas for data imputation or further investigation. Additionally, 'Results First Posted' has a very low non-null count (36), indicating that this information is seldom available.


-->Study Characteristics:
Categorical columns such as 'Status', 'Gender', 'Study Type', and 'Study Designs' convey various attributes of studies, including their status, gender inclusion criteria, study type, and design. The 'Age' column presumably denotes the target age group for each study.


-->Temporal Data:
Date-related columns like 'Start Date', 'Primary Completion Date', 'Completion Date', 'First Posted', and 'Last Update Posted' imply temporal aspects of the studies. However, they are currently stored as object data types and may necessitate conversion for temporal analysis.


-->Enrollment Data:
The 'Enrollment' column furnishes information about the number of participants enrolled in each study, with enrollment figures ranging from 0 to 20 million. The wide range suggests differing study sizes.


-->URL and Identifiers:
'NCT Number' acts as a unique identifier for each study, while the 'Other IDs' column contains supplementary identifiers. Additionally, the 'URL' column potentially provides links to further information about each study.


-->Locations and Collaborators:
The 'Locations' column indicates the geographical locations where studies are conducted, whereas 'Sponsor/Collaborators' offers insights into the entities sponsoring or collaborating on the studies.

In [ ]:
df[["Age"]]

Age
0        18 Years and older   (Adult, Older Adult)
1        18 Years and older   (Adult, Older Adult)
2        18 Years and older   (Adult, Older Adult)
3                        Child, Adult, Older Adult
4                   18 Years to 48 Years   (Adult)
...                                            ...
5778  21 Years to 100 Years   (Adult, Older Adult)
5779         3 Months to 18 Years   (Child, Adult)
5780                18 Years to 64 Years   (Adult)
5781          8 Years to 20 Years   (Child, Adult)
5782     18 Years and older   (Adult, Older Adult)

[5783 rows x 1 columns]

# Question 03
**Loc and iloc**

** Select all columns for the first clinical trial in the dataset.**








In [ ]:
df.head(1).T

0
Rank                                                                     1
NCT Number                                                     NCT04785898
Title                    Diagnostic Performance of the ID Now™ COVID-19...
Acronym                                                        COVID-IDNow
Status                                              Active, not recruiting
Study Results                                         No Results Available
Conditions                                                         Covid19
Interventions             Diagnostic Test: ID Now™ COVID-19 Screening Test
Outcome Measures         Evaluate the diagnostic performance of the ID ...
Sponsor/Collaborators                Groupe Hospitalier Paris Saint Joseph
Gender                                                                 All
Age                              18 Years and older   (Adult, Older Adult)
Phases                                                      Not Applicable
Enrollment                                                          1000.0
Funded Bys                                                           Other
Study Type                                                  Interventional
Study Designs            Allocation: N/A|Intervention Model: Single Gro...
Other IDs                                                      COVID-IDNow
Start Date                                                November 9, 2020
Primary Completion Date                                  December 22, 2020
Completion Date                                             April 30, 2021
First Posted                                                 March 8, 2021
Results First Posted                                                   NaN
Last Update Posted                                           March 8, 2021
Locations                Groupe Hospitalier Paris Saint-Joseph, Paris, ...
Study Documents                                                        NaN
URL                            https://ClinicalTrials.gov/show/NCT04785898

In [ ]:
df.head(1)

Rank   NCT Number                                              Title  \
0     1  NCT04785898  Diagnostic Performance of the ID Now™ COVID-19...   

       Acronym                  Status         Study Results Conditions  \
0  COVID-IDNow  Active, not recruiting  No Results Available    Covid19   

                                      Interventions  \
0  Diagnostic Test: ID Now™ COVID-19 Screening Test   

                                    Outcome Measures  \
0  Evaluate the diagnostic performance of the ID ...   

                   Sponsor/Collaborators  ...    Other IDs        Start Date  \
0  Groupe Hospitalier Paris Saint Joseph  ...  COVID-IDNow  November 9, 2020   

  Primary Completion Date  Completion Date   First Posted  \
0       December 22, 2020   April 30, 2021  March 8, 2021   

  Results First Posted Last Update Posted  \
0                  NaN      March 8, 2021   

                                           Locations Study Documents  \
0  Groupe Hospitalier Paris Saint-Joseph, Paris, ...             NaN   

                                           URL  
0  https://ClinicalTrials.gov/show/NCT04785898  

[1 rows x 27 columns]

In [ ]:
df.loc[0]

Rank                                                                       1
NCT Number                                                       NCT04785898
Title                      Diagnostic Performance of the ID Now™ COVID-19...
Acronym                                                          COVID-IDNow
Status                                                Active, not recruiting
Study Results                                           No Results Available
Conditions                                                           Covid19
Interventions               Diagnostic Test: ID Now™ COVID-19 Screening Test
Outcome Measures           Evaluate the diagnostic performance of the ID ...
Sponsor/Collaborators                  Groupe Hospitalier Paris Saint Joseph
Gender                                                                   All
Age                                18 Years and older   (Adult, Older Adult)
Phases                                                        Not Applicable

** Retrieve the Title and Status of the clinical trial with the NCT Number'NCT04595136'.**

In [ ]:
trail_info = df.loc[df["NCT Number"] == "NCT04595136", ["Title","Status"]]
trail_info

Title              Status
1  Study to Evaluate the Efficacy of COVID19-0001...  Not yet recruiting

** Get the Sponsor/Collaborators and Start Date for clinical trials that areRecruiting.**

In [ ]:
df.loc[df["Status"]=="Recruiting", ["Sponsor/Collaborators", "Start Date"]]

Sponsor/Collaborators         Start Date
2                          University of Milano Bicocca        May 7, 2020
4     Centre Hospitalier Régional d'Orléans|Centre d...        May 5, 2020
9     Fundacion Arturo Lopez Perez|Confederación de ...      April 7, 2020
10                                  The Christ Hospital     April 28, 2020
11                         University of Milano Bicocca     April 16, 2020
...                                                 ...                ...
5776  University of Pittsburgh|U.S. National Science...   October 26, 2020
5777                               University of Zurich    August 25, 2020
5778  University of Wisconsin, Madison|National Inst...     March 23, 2020
5780         University of Illinois at Urbana-Champaign  February 10, 2021
5781  Mental Health Services in the Capital Region, ...     March 22, 2021

[2805 rows x 2 columns]

** Select the first 5 rows and columns Title, Conditions, and Outcome Measures.**

In [ ]:
df.head(1)

Rank   NCT Number                                              Title  \
0     1  NCT04785898  Diagnostic Performance of the ID Now™ COVID-19...   

       Acronym                  Status         Study Results Conditions  \
0  COVID-IDNow  Active, not recruiting  No Results Available    Covid19   

                                      Interventions  \
0  Diagnostic Test: ID Now™ COVID-19 Screening Test   

                                    Outcome Measures  \
0  Evaluate the diagnostic performance of the ID ...   

                   Sponsor/Collaborators  ...    Other IDs        Start Date  \
0  Groupe Hospitalier Paris Saint Joseph  ...  COVID-IDNow  November 9, 2020   

  Primary Completion Date  Completion Date   First Posted  \
0       December 22, 2020   April 30, 2021  March 8, 2021   

  Results First Posted Last Update Posted  \
0                  NaN      March 8, 2021   

                                           Locations Study Documents  \
0  Groupe Hospitalier Paris Saint-Joseph, Paris, ...             NaN   

                                           URL  
0  https://ClinicalTrials.gov/show/NCT04785898  

[1 rows x 27 columns]

In [ ]:
selected_data = df.loc[:4, ['Title', 'Conditions', 'Outcome Measures']]

selected_data

Title  \
0  Diagnostic Performance of the ID Now™ COVID-19...   
1  Study to Evaluate the Efficacy of COVID19-0001...   
2  Lung CT Scan Analysis of SARS-CoV2 Induced Lun...   
3  The Role of a Private Hospital in Hong Kong Am...   
4         Maternal-foetal Transmission of SARS-Cov-2   

                                          Conditions  \
0                                            Covid19   
1                               SARS-CoV-2 Infection   
2                                            covid19   
3                                              COVID   
4  Maternal Fetal Infection Transmission|COVID-19...   

                                    Outcome Measures  
0  Evaluate the diagnostic performance of the ID ...  
1  Change on viral load results from baseline aft...  
2  A qualitative analysis of parenchymal lung dam...  
3  Proportion of asymptomatic subjects|Proportion...  
4  COVID-19 by positive PCR in cord blood and / o...

** Find the Completion Date and URL for the last 3 clinical trials in the dataset.**

In [ ]:
df.columns

Index(['Rank', 'NCT Number', 'Title', 'Acronym', 'Status', 'Study Results',
       'Conditions', 'Interventions', 'Outcome Measures',
       'Sponsor/Collaborators', 'Gender', 'Age', 'Phases', 'Enrollment',
       'Funded Bys', 'Study Type', 'Study Designs', 'Other IDs', 'Start Date',
       'Primary Completion Date', 'Completion Date', 'First Posted',
       'Results First Posted', 'Last Update Posted', 'Locations',
       'Study Documents', 'URL'],
      dtype='object')

In [ ]:
df.columns.isin(['Completion Date', 'URL'])

array([False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False,  True, False, False, False, False, False,  True])

# Question 4
**Determine the missing values in the whole dataset and analyze missing values in each
column.**

In [ ]:
df.isnull().sum()

Rank                          0
NCT Number                    0
Title                         0
Acronym                    3303
Status                        0
Study Results                 0
Conditions                    0
Interventions               886
Outcome Measures             35
Sponsor/Collaborators         0
Gender                       10
Age                           0
Phases                     2461
Enrollment                   34
Funded Bys                    0
Study Type                    0
Study Designs                35
Other IDs                     1
Start Date                   34
Primary Completion Date      36
Completion Date              36
First Posted                  0
Results First Posted       5747
Last Update Posted            0
Locations                   585
Study Documents            5601
URL                           0
dtype: int64

**Calculate the sum of duplicate rows**

In [ ]:
df[df.duplicated()].sum()

Rank                         0
NCT Number                   0
Title                        0
Acronym                      0
Status                       0
Study Results                0
Conditions                   0
Interventions                0
Outcome Measures             0
Sponsor/Collaborators        0
Gender                       0
Age                          0
Phases                       0
Enrollment                 0.0
Funded Bys                   0
Study Type                   0
Study Designs                0
Other IDs                    0
Start Date                   0
Primary Completion Date      0
Completion Date              0
First Posted                 0
Results First Posted         0
Last Update Posted           0
Locations                    0
Study Documents              0
URL                          0
dtype: object

In [ ]:
df.duplicated().sum()

0

# Question 6: -
**Solve following question by using conditional statements**


** How many studies have an enrollment greater than a certain threshold?**

In [ ]:
threshold = 170

(df["Enrollment"] > threshold).value_counts()

Enrollment
False    2909
True     2874
Name: count, dtype: int64

In [ ]:
great_th_thresh = df["Enrollment"] > threshold
great_th_thresh

0        True
1       False
2        True
3        True
4       False
        ...  
5778     True
5779     True
5780    False
5781     True
5782     True
Name: Enrollment, Length: 5783, dtype: bool

In [ ]:
df_great_thre = df[great_th_thresh]
df_great_thre

Rank   NCT Number                                              Title  \
0        1  NCT04785898  Diagnostic Performance of the ID Now™ COVID-19...   
2        3  NCT04395482  Lung CT Scan Analysis of SARS-CoV2 Induced Lun...   
3        4  NCT04416061  The Role of a Private Hospital in Hong Kong Am...   
6        7  NCT04476940  COVID-19 Breastfeeding Guideline for African-A...   
7        8  NCT04634214  The Severity of COVID 19 in Diabetes and Non-d...   
...    ...          ...                                                ...   
5777  5778  NCT04574466  Scaling-up Psychological Interventions With Sy...   
5778  5779  NCT04011644  Mobile Health for Alcohol Use Disorders in Cli...   
5779  5780  NCT04681339  Antibiotic Prescription in Children Hospitaliz...   
5781  5782  NCT04804917           3-year Follow-up of the Mind My Mind RCT   
5782  5783  NCT04680000  Chronic Pain Management In Primary Care Using ...   

           Acronym                  Status         Study Results  \
0      COVID-IDNow  Active, not recruiting  No Results Available   
2      TAC-COVID19              Recruiting  No Results Available   
3         COVID-19  Active, not recruiting  No Results Available   
6         COVID-BF      Not yet recruiting  No Results Available   
7          COVID19      Not yet recruiting  No Results Available   
...            ...                     ...                   ...   
5777           NaN              Recruiting  No Results Available   
5778           NaN              Recruiting  No Results Available   
5779           NaN      Not yet recruiting  No Results Available   
5781  MindMyMindFU              Recruiting  No Results Available   
5782           NaN      Not yet recruiting  No Results Available   

                                             Conditions  \
0                                               Covid19   
2                                               covid19   
3                                                 COVID   
6                       Covid19|Exclusive Breastfeeding   
7                                Covid19|Type2 Diabetes   
...                                                 ...   
5777  Distress|PTSD|Anxiety|Depression|Trauma|Functi...   
5778                      Alcohol Drinking|Telemedicine   
5779  Community Acquired Pneumonia in Children|Antib...   
5781  Emotional Problem|Anxiety Disorder of Childhoo...   
5782                                       Chronic Pain   

                                          Interventions  \
0      Diagnostic Test: ID Now™ COVID-19 Screening Test   
2     Other: Lung CT scan analysis in COVID-19 patients   
3             Diagnostic Test: COVID 19 Diagnostic Test   
6            Behavioral: COVID-19 Breastfeeding Support   
7                                                   NaN   
...                                                 ...   
5777                Behavioral: Problem Management Plus   
5778  Behavioral: A-CHESS self-monitored|Behavioral:...   
5779  Other: Antibiotic treatment|Other: No antibiot...   
5781                                                NaN   
5782  Behavioral: Brief Cognitive Behavioral Therapy...   

                                       Outcome Measures  \
0     Evaluate the diagnostic performance of the ID ...   
2     A qualitative analysis of parenchymal lung dam...   
3     Proportion of asymptomatic subjects|Proportion...   
6     COVID-19 breastfeeding guidance adherence at b...   
7     Severity of COVID 19 among people with and wit...   
...                                                 ...   
5777  Change in psychological distress|Change in pos...   
5778  Number of risky drinking days|Number of patien...   
5779  Antibiotic treatment rates in hospitalized chi...   
5781  The child's impact of mental health problems r...   
5782  Defense and Veterans Pain Rating Scale (DVPRS)...   

                                  Sponsor/Collaborators  ...  \
0                 Groupe Hospitalier Paris Saint Joseph  ... 

In [ ]:
count_studies_above_threshold = df_great_thre.shape[0]
count_studies_above_threshold

2874

In [ ]:
print("Number of studies with enrollment greater than", threshold, ":", count_studies_above_threshold)

Number of studies with enrollment greater than 170 : 2874


**How many clinical trials have 'No Results Available' ?**  

In [ ]:
(df["Study Results"] == "No Results Available").value_counts()

Study Results
True     5747
False      36
Name: count, dtype: int64

In [ ]:
# to find the numbers of trails have 'No Results Available'
df[df["Study Results"] == "No Results Available"].shape[0]

5747

**How many clinical trials are in an "Completed" and "Recruiting" status?**

In [ ]:
clinical_trails = (df["Status"] == "Completed") | (df["Status"] == "Recruiting")
clinical_trails

0       False
1       False
2        True
3       False
4        True
        ...  
5778     True
5779    False
5780     True
5781     True
5782    False
Name: Status, Length: 5783, dtype: bool

In [ ]:
#  clinical trials are in an "Completed" and "Recruiting" status..
df[clinical_trails].shape[0]

3830

**How many clinical trials are related to 'COVID-19'?**

In [ ]:
df[df["Acronym"] == "COVID-19"].shape[0]

47

In [ ]:
df[df["Conditions"] == "COVID-19"].shape[0]

720

In [ ]:
count_covid_trials = df[df['Conditions'].str.contains('COVID-19', case=False, na=False)].shape[0]

print("Nmbr of clinical_trials related to COVID-19:", count_covid_trials)

Nmbr of clinical_trials related to COVID-19: 1872


**How many clinical trials started after January 1, 2020?**

In [ ]:
df["Start Date"]

0        November 9, 2020
1        November 2, 2020
2             May 7, 2020
3            May 25, 2020
4             May 5, 2020
              ...        
5778       March 23, 2020
5779           April 2021
5780    February 10, 2021
5781       March 22, 2021
5782        February 2021
Name: Start Date, Length: 5783, dtype: object

In [ ]:
df['Start Date'] = pd.to_datetime(df['Start Date'], errors='coerce')
df['Start Date']

0      2020-11-09
1      2020-11-02
2      2020-05-07
3      2020-05-25
4      2020-05-05
          ...    
5778   2020-03-23
5779          NaT
5780   2021-02-10
5781   2021-03-22
5782          NaT
Name: Start Date, Length: 5783, dtype: datetime64[ns]

In [ ]:
df[df['Start Date'] > pd.Timestamp('2020-01-01')].shape

(5066, 27)

In [ ]:
# Convert 'Start Date' column to datetime format, handling different date formats
df['Start Date'] = pd.to_datetime(df['Start Date'], errors='coerce')

# Filter trials started after January 1, 2020
count_trials_after_2020 = df[df['Start Date'] > pd.Timestamp('2020-01-01')].shape[0]

print("Number of clinical_trials after_ January 1, 2020:", count_trials_after_2020)

Number of clinical_trials after_ January 1, 2020: 5066


**<--<*>/<*>- -<*>/<*>--<*>/<*>--<*>/<*>--<*>/<*>--<*>/<*>--<*>/<*>--<*>/<*>--<*>/<*>--<*>/<*>--<*>/<*>--<*>/<*>--<*>/<*>--<*>/<*>--<*>/<*>-->**